### ISSUE TITLE
Can't run on Mac using nzbhydra2wrapper.py

### ISSUE THREAD

**Issue by paulrumens** [theotherp/nzbhydra2#202]
I have installed the Java 6 Runtime from Apple:
https://support.apple.com/kb/DL1572?locale=en_US

When I try and run the wrapper I get: 

server:~ user$ python /Users/user/Desktop/nzbhydra2-1/nzbhydra2wrapper.py
Logging wrapper output to /Users/user/Desktop/nzbhydra2-1/data/logs/wrapper.log
2018-06-24 16:51:51,533 INFO - No file nzbhydra.yml found. Using 128M XMX
2018-06-24 16:51:51,534 INFO - Starting NZBHydra main process with command line: java -Xmx128M -DfromWrapper -XX:TieredStopAtLevel=1 -noverify -Xloggc:/Users/user/Desktop/nzbhydra2-1/data/logs/gclog-2018-06-24_16-51-51.log -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintTenuringDistribution -XX:+PrintGCCause -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=10 -XX:GCLogFileSize=5M -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/Users/user/Desktop/nzbhydra2-1/data/logs -Dspring.output.ansi.enabled=ALWAYS -jar /Users/user/Desktop/nzbhydra2-1/lib/core-1.5.1-exec.jar --datafolder /Users/user/Desktop/nzbhydra2-1/data in folder /Users/user/Desktop/nzbhydra2-1
Unrecognized VM option '+PrintGCCause'
Could not create the Java virtual machine.
2018-06-24 16:51:51,651 ERROR - Main process shut down unexpectedly. If the wrapper was started in daemon mode you might not see the error output. Start Hydra manually with the same parameters in the same environment to see it

What version of Java should I be trying?

**Comment by theotherp** [comment]
You need Java 8 (u101 or higher).

**Comment by paulrumens** [comment]
OK I downloaded Java 8 u171 here is the error:

server:~ user$ python /Applications/nzbhydra2/nzbhydra2wrapper.py 
Logging wrapper output to /Applications/nzbhydra2/data/logs/wrapper.log
2018-06-24 19:37:18,122  INFO - No file nzbhydra.yml found. Using 128M XMX
2018-06-24 19:37:18,122  INFO - Starting NZBHydra main process with command line: java -Xmx128M -DfromWrapper -XX:TieredStopAtLevel=1 -noverify -Xloggc:/Applications/nzbhydra2/data/logs/gclog-2018-06-24_19-37-18.log -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintTenuringDistribution -XX:+PrintGCCause -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=10 -XX:GCLogFileSize=5M -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/Applications/nzbhydra2/data/logs -Dspring.output.ansi.enabled=ALWAYS -jar /Applications/nzbhydra2/lib/core-1.5.1-exec.jar --datafolder /Applications/nzbhydra2/data in folder /Applications/nzbhydra2
Unrecognized VM option '+PrintGCCause'
Could not create the Java virtual machine.
2018-06-24 19:37:18,178  ERROR - Main process shut down unexpectedly. If the wrapper was started in daemon mode you might not see the error output. Start Hydra manually with the same parameters in the same environment to see it

**Comment by theotherp** [comment]
Huh, no idea. You'll have to wait until Wednesday when I'm back home,
sorry.

**Comment by theotherp** [comment]
Please run `java -version` and post the output.

**Comment by theotherp** [comment]
That means that the default java version is still 6. You need to change it to 8. I found this link: https://alvinalexander.com/macos/how-to-set-mac-osx-java-version-path with a bit more explanation here: https://stackoverflow.com/questions/21964709/how-to-set-or-change-the-default-java-jdk-version-on-os-x

**Comment by paulrumens** [comment]
OK Java 8 wasn't even showing up. After some messing around, I have it working (I have a whole new issue but I will open a new case for that!), I will give a rough outline incase any other Mac users come here!

Run this
_/usr/libexec/java_home -V_

If you only have Java 6, read on!

Turns out the Oracle installer doesn't do a very good job, I uninstalled it using this

_sudo rm -fr /Library/Internet\ Plug-Ins/JavaAppletPlugin.plugin 
sudo rm -fr /Library/PreferencePanes/JavaControlPanel.prefPane 
sudo rm -fr ~/Library/Application\ Support/Oracle/Java_

Next I use Homebrew to install it. You need the Xcode so:
_xcode-select --install_

Now you can install Homebrew
_ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"_

And finally 
_brew cask install java8_

Now double check that you have Java 8
_/usr/libexec/java_home -V_

Now lets see if you can run Hydra (replace with your url for Hydra)
python Applications/NZBHydra/nzbhydra2wrapper.py

But you dont want to go to terminal everytime you want to start Hydra, so grab a copy of BBEdit
https://www.barebones.com/products/bbedit/

Create a new document and enter this (again, replace with your url for Hydra)

```
<plist version='1.0'>
<dict>
 <key>Label</key>
 <string>nzbhydra</string>
 <key>ProgramArguments</key>
 <array>
 <string>/usr/bin/python2.7</string>
 <string>/Applications/NZBHydra/nzbhydra2wrapper.py</string>
 </array>
 <key>KeepAlive</key>
 <true/>
</dict>
</plist>
```


Save this as nzbhdra.plist and drop it into /Library/LaunchDaemons/

Reboot

**Comment by theotherp** [comment]
Hey, thanks for the detailed instructions.

Do I understand this correctly that your issue is resolved?

**Comment by paulrumens** [comment]
Yep everything working now. I will fling some bitcoin your way soon!
